api/src/test/java/com/netflix/iceberg/transforms/TestProjection.java

```diff
@@ -86,6 +87,53 @@ public void testIdentityProjection() {
     }
   }
 
+  @Test
+  public void testCaseInsensitiveIdentityProjection() {
+    List<UnboundPredicate<?>> predicates = Lists.newArrayList(
+            Expressions.notNull("ID"),
+            Expressions.isNull("ID"),
+            Expressions.lessThan("ID", 100),
+            Expressions.lessThanOrEqual("ID", 101),
+            Expressions.greaterThan("ID", 102),
+            Expressions.greaterThanOrEqual("ID", 103),
+            Expressions.equal("ID", 104),
+            Expressions.notEqual("ID", 105)
+    );
+
+    PartitionSpec spec = PartitionSpec.builderFor(SCHEMA)
+            .identity("id")
+            .build();
+
+    for (UnboundPredicate<?> predicate : predicates) {
+      // get the projected predicate
+      Expression expr = Projections.inclusive(spec, false).project(predicate);
+      UnboundPredicate<?> projected = assertAndUnwrapUnbound(expr);
+
+      // check inclusive the bound predicate to ensure the types are correct
+      BoundPredicate<?> bound = assertAndUnwrap(predicate.bind(spec.schema().asStruct(), false));
+
+      Assert.assertEquals("Field name should match partition struct field",
+              "id", projected.ref().name());
+      Assert.assertEquals("Operation should match", bound.op(), projected.op());
+
+      if (bound.literal() != null) {
+        Assert.assertEquals("Literal should be equal",
+                bound.literal().value(), projected.literal().value());
+      } else {
+        Assert.assertNull("Literal should be null", projected.literal());
+      }
+    }
+  }
+
+  @Test(expected = ValidationException.class)
+  public void testCaseSensitiveIdentityProjection() {
```

Comment:
Using `TestHelpers` would allow you to use the same structure as the test above and test all of the expressions.

Reply:
Agreed.